tests/ignite/handlers/test_ema_handler.py

```diff
@@ -0,0 +1,319 @@
+    assert hasattr(engine.state, "ema_momentum_1")
+
+    # handler_2 update EMA model for model_2 every 2 iterations
+    ema_handler_2.attach(engine, "ema_momentum_2", event=Events.ITERATION_COMPLETED(every=2))
+    assert hasattr(engine.state, "ema_momentum_2")
+
+    # engine will run 4 iterations
+    engine.run(range(2), max_epochs=2)
+    ema_weight_1 = ema_handler_1.ema_model.weight.data
+    ema_weight_2 = ema_handler_2.ema_model.weight.data
+    torch.testing.assert_allclose(ema_weight_1, torch.full((1, 2), 4.0625))
+    torch.testing.assert_allclose(ema_weight_2, torch.full((1, 2), 3.5))
+
+    assert engine.state.ema_momentum_1 == 0.5
+    assert engine.state.ema_momentum_2 == 0.5
+
+    model_3 = get_dummy_model()
+    ema_handler_3 = EMAHandler(model_3)
+    with pytest.raises(ValueError, match="Please select another name"):
+        ema_handler_3.attach(engine, "ema_momentum_2")
+
+
+def _test_ema_final_weight(model, device, ddp=False, interval=1):
+    """Test if final smoothed weights are correct"""
+    if isinstance(device, str):
+        device = torch.device(device)
+    model = model.to(device)
+    if ddp:
+        model = idist.auto_model(model)
+    step_fn = _get_dummy_step_fn(model)
+    engine = Engine(step_fn)
+
+    # momentum will be constantly 0.5
+    ema_handler = EMAHandler(model, momentum_warmup=0.5, momentum=0.5, warmup_iters=1)
+    ema_handler.attach(engine, "model", event=Events.ITERATION_COMPLETED(every=interval))
+
+    # engine will run 4 iterations
+    engine.run(range(2), max_epochs=2)
+
+    ema_weight = ema_handler.ema_model.weight.data
+    model_weight = model.weight.data
+    assert ema_weight.device == device
+    assert model_weight.device == device
+    if interval == 1:
+        torch.testing.assert_allclose(ema_weight, torch.full((1, 2), 4.0625, device=device))
+    elif interval == 2:
+        torch.testing.assert_allclose(ema_weight, torch.full((1, 2), 3.5, device=device))
+    else:
+        pass
+    torch.testing.assert_allclose(model_weight, torch.full((1, 2), 5.0, device=device))
+
+
+@pytest.mark.parametrize("interval", [1, 2])
+def test_ema_final_weight_cpu(get_dummy_model, interval):
+    device = torch.device("cpu")
+    _test_ema_final_weight(get_dummy_model(), device=device, ddp=False, interval=interval)
+
+
+@pytest.mark.parametrize("interval", [1, 2])
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="Skip if no GPU")
+def test_ema_final_weight_cuda(get_dummy_model, interval):
+    device = torch.device("cuda:0")
+    _test_ema_final_weight(get_dummy_model(), device=device, ddp=False, interval=interval)
+
+
+@pytest.mark.distributed
+@pytest.mark.parametrize("interval", [1, 2])
+@pytest.mark.skipif(not idist.has_native_dist_support, reason="Skip if no native dist support")
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="Skip if no GPU")
+def test_ema_final_weight_nccl_cuda(get_dummy_model, distributed_context_single_node_nccl, interval):
```

Comment:
Please, remove `cuda` from the test name here (otherwise this test is also running with Windows and failing as no NCCL)
```diff
- test_ema_final_weight_nccl_cuda
+ test_ema_final_weight_nccl
```